samtranslator/model/sam_resources.py

```diff
@@ -2144,26 +2147,76 @@ class SamGraphQLApi(SamResourceMacro):
     Name: Optional[str]
     SchemaInline: Optional[str]
     SchemaUri: Optional[str]
+    Logging: Optional[Union[Dict[str, Any], bool]]
 
     @cw_timer
     def to_cloudformation(self, **kwargs: Any) -> List[Resource]:
-        appsync_api = self._construct_appsync_api()
+        appsync_api, cloudwatch_role = self._construct_appsync_api_resources()
         appsync_schema = self._construct_appsync_schema(appsync_api.get_runtime_attr("api_id"))
+
         resources: List[Resource] = [appsync_api, appsync_schema]
 
+        if cloudwatch_role:
+            resources.append(cloudwatch_role)
+
         return resources
 
-    def _construct_appsync_api(self) -> GraphQLApi:
+    def _construct_appsync_api_resources(self) -> Tuple[GraphQLApi, Optional[IAMRole]]:
         api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
 
         api.AuthenticationType = sam_expect(self.Auth.get("Type"), self.logical_id, "Auth.Type").to_be_a_string()
         api.Name = self.Name if self.Name else self.logical_id
         api.XrayEnabled = self.XrayEnabled
 
+        cloudwatch_role = None  # conditionally generated, if left as None then its not added to resources
+
+        # Logging has 3 possible types: dict, bool, and None.
+        # GraphQLApi will not include logging if and only if the user explicity sets Logging as false boolean.
+        # It will for every other value (including true boolean which is essentially same as None).
+        if type(self.Logging) != bool or self.Logging is True:
+            api.LogConfig = self._parse_logging_properties()
+
+            # We create a CloudWatch role for the user if the Logging property is not a dictionary
+            # that contains the "CloudWatchLogsRoleArn" property.
+            if type(self.Logging) != dict or "CloudWatchLogsRoleArn" not in self.Logging:
+                cloudwatch_role = self._construct_cloudwatch_role()
+
         if self.Tags:
             api.Tags = get_tag_list(self.Tags)
 
-        return api
+        return api, cloudwatch_role
+
+    def _parse_logging_properties(self) -> LogConfigType:
+        log_config: LogConfigType = {}
+
+        if type(self.Logging) == dict and "CloudWatchLogsRoleArn" in self.Logging:
```

Comment:
```suggestion
        if isinstance(self.Logging, dict):
           if "CloudWatchLogsRoleArn" in self.Logging:
           ....
           if "ExcludeVerboseContent" in self.Logging:
           ...
           if "FieldLogLevel" in self.Logging
           ....
```

Reply:
Yep, I'll add this change as well

Reply:
While trying to refactor this, I realized we can’t simply abstract away this condition into its own check.
There is two cases where the key should be defaulted in Logging:
- Logging is a Bool / None type
- Logging is a dictionary, but does not contain a key for the defaulted property.

We could have 2 separate if statements, one to handle when Logging is Bool/None, and one where it’s a dictionary, but there would be even more duplicated code and I’m not convinced that’s better clarity wise. So I’m leaning towards keeping this implementation, but replacing `type` with `isinstance`.